File: mobile/ios/HiAir/DesignSystem/TimeOfDayBackground.swift
import SwiftUI

enum TimeOfDayBackground {
    static func gradient(for date: Date = Date()) -> LinearGradient {
        LinearGradient(
            colors: [
                HiAirColors.Surface.bg0,
                TimeOfDayPhase.from(date: date).colors.last ?? HiAirColors.Surface.bg2,
            ],
            startPoint: .top,
            endPoint: .bottom
        )
    }

    static func surfacePrimary(for date: Date = Date()) -> Color {
        _ = date
        return HiAirColors.Surface.bg2
    }

    static func surfaceSecondary(for date: Date = Date()) -> Color {
        _ = date
        return HiAirColors.Surface.bg3
    }

    static func surfaceElevated(for date: Date = Date()) -> Color {
        _ = date
        return HiAirColors.Surface.bg3.lightened(by: 0.08)
    }
}

File: mobile/android/app/src/main/java/com/hiair/ui/design/TimeOfDayBackground.kt
package com.hiair.ui.design

import android.graphics.drawable.GradientDrawable

object TimeOfDayBackground {
    fun pageGradient(): GradientDrawable {
        val phase = Tokens.TimeOfDayPhase.current()
        return GradientDrawable(
            GradientDrawable.Orientation.TOP_BOTTOM,
            intArrayOf(phase.top, phase.bottom)
        )
    }

    fun surfacePrimary(): Int = HiAirColors.Surface.bg2

    fun surfaceSecondary(): Int = HiAirColors.Surface.bg3

    fun surfaceElevated(): Int = lighten(HiAirColors.Surface.bg3, 0.08f)

    private fun lighten(color: Int, amount: Float): Int {
        val hsv = FloatArray(3)
        android.graphics.Color.colorToHSV(color, hsv)
        hsv[2] = (hsv[2] + amount).coerceAtMost(1f)
        return android.graphics.Color.HSVToColor(android.graphics.Color.alpha(color), hsv)
    }
}
